src/securify/next_free_index_assigner.rs

```diff
@@ -0,0 +1,194 @@
+#![allow(clippy::type_complexity)]
+
+use crate::prelude::*;
+
+use rand::Rng;
+use sha2::{Digest, Sha256, Sha512};
+
+#[derive(Clone, Copy, PartialEq)]
+pub struct NextFreeIndexAssignerRequest<'p> {
+    pub key_space: KeySpace,
+    pub entity_kind: CAP26EntityKind,
+    pub factor_source_id: FactorSourceIDFromHash,
+    pub profile: &'p Profile,
+    pub network_id: NetworkID,
+}
+
+pub struct NextFreeIndexAssigner {
```

Comment:
💡 This is MUCH simplified. A proper one would be async, and use `pre_derived_keys_cache` and gateway for `IEI`!